x-pack/test/alerting_api_integration/security_and_spaces/group2/tests/actions/get_all_system.ts

```diff
@@ -0,0 +1,529 @@
+                },
+                {
+                  connector_type_id: 'test.system-action-connector-adapter',
+                  id: 'system-connector-test.system-action-connector-adapter',
+                  is_deprecated: false,
+                  is_preconfigured: false,
+                  is_system_action: true,
+                  name: 'System action: test.system-action-connector-adapter',
+                  referenced_by_count: 0,
+                },
+                {
+                  connector_type_id: 'test.system-action-kibana-privileges',
+                  id: 'system-connector-test.system-action-kibana-privileges',
+                  is_deprecated: false,
+                  is_preconfigured: false,
+                  is_system_action: true,
+                  name: 'System action: test.system-action-kibana-privileges',
+                  referenced_by_count: 0,
+                },
+                {
+                  id: 'custom-system-abc-connector',
+                  is_preconfigured: true,
+                  is_system_action: false,
+                  is_deprecated: false,
+                  connector_type_id: 'system-abc-action-type',
+                  name: 'SystemABC',
+                  referenced_by_count: 0,
+                },
+                {
+                  id: 'preconfigured.test.index-record',
+                  is_preconfigured: true,
+                  is_system_action: false,
+                  is_deprecated: false,
+                  connector_type_id: 'test.index-record',
+                  name: 'Test:_Preconfigured_Index_Record',
+                  referenced_by_count: 0,
+                },
+                {
+                  id: 'my-test-email',
+                  is_preconfigured: true,
+                  is_system_action: false,
+                  is_deprecated: false,
+                  connector_type_id: '.email',
+                  name: 'TestEmail#xyz',
+                  referenced_by_count: 0,
+                },
+              ]);
+              break;
+            default:
+              throw new Error(`Scenario untested: ${JSON.stringify(scenario)}`);
+          }
+        });
+
+        it('should handle get all request appropriately with proper referenced_by_count', async () => {
+          const { body: createdAction } = await supertest
+            .post(`${getUrlPrefix(space.id)}/api/actions/connector`)
+            .set('kbn-xsrf', 'foo')
+            .send({
+              name: 'My action',
+              connector_type_id: 'test.index-record',
+              config: {
+                unencrypted: `This value shouldn't get encrypted`,
+              },
+              secrets: {
+                encrypted: 'This value should be encrypted',
+              },
+            })
+            .expect(200);
+          objectRemover.add(space.id, createdAction.id, 'action', 'actions');
+
+          const { body: createdAlert } = await supertest
```

Comment:
should we update this test to use system actions in the rule to make sure the `referenced_by_count` for system actions is calculated correctly?

Reply:
We do not persist the ID of the system action to the references array. Same as we do for preconfigured connectors. So the `referenced_by_count` is always zero. I added a test to ensure that.